app/scripts/controllers/token-rates.js

```diff
@@ -26,32 +26,22 @@ class TokenRatesController {
   async updateExchangeRates () {
     if (!this.isActive) { return }
     const contractExchangeRates = {}
-    // copy array to ensure its not modified during iteration
-    const tokens = this._tokens.slice()
-    for (const token of tokens) {
-      if (!token) return log.error(`TokenRatesController - invalid tokens state:\n${JSON.stringify(tokens, null, 2)}`)
-      const address = token.address
-      contractExchangeRates[address] = await this.fetchExchangeRate(address)
+    const pairs = this._tokens.map(token => `pairs[]=${token.address}/ETH`)
```

Comment:
Not a 100% sure about this but since we support any base currency now (under settings). Shouldn't we use the selected currency instead of hardcoding ETH?

Reply:
Good catch. Looks like our token pricing has been broken since we landed that functionality (if the user changed their native currency from ETH.)